### PR TITLE
ci: add run clippy with --tests

### LINKED ISSUE #105: ci - cargo clippy should be run with --tests
Otherwise tests aren't covered.

### PR DESCRIPTION
**Pull request by PsiACE**
**Summary**

- toolchain: add components
- ci: add run clippy with --tests
- docs: update the contributing doc

**Related**

Fixes #105

### CODE REVIEW

File: .github/workflows/checkcode.yml [hunk 1]
```diff
@@ -78,7 +78,7 @@ jobs:
       uses: actions-rs/cargo@v1
       with:
         command: clippy
-        args: -- -D warnings
+        args: --tests -- -D warnings
```
Comment (tisonkun) [inline .github/workflows/checkcode.yml:81]:
```suggestion
        args: --workspace --tests --all-features -- -D warnings
```

To avoid further fixing :P

**Review by tisonkun — CHANGES_REQUESTED** [review]
This patch may still miss internal crates and features topics.

Also, you can see the comment https://github.com/engula/engula/issues/105#issuecomment-968075641

**Review by tisonkun — APPROVED** [review]
LGTM.

**Review by huachaohuang — APPROVED** [review]
LGTM, thanks.

**Pull request merged**